common/models/templates/src/trie.ts

```diff
@@ -128,20 +133,22 @@ export class TrieTraversal implements LexiconTraversal {
   // Handles one code unit at a time.
   private _child(char: USVString): TrieTraversal | undefined {
     const root = this.root;
+    const toKey = this.toKey;
```

Comment:
Why copy the class variable if it isn't going to change?

Reply:
Two more references exist later on within the function, making three total.  I've generally preferred to avoid constant `this.` prefixing for the same thing repeatedly when I can help it.

Also, though but a minor point, in my experience `esbuild` doesn't minify things accessed by property - it won't minify `this.toKey`.  On the other hand a local constant variable (like our copied `toKey` variable) is perfectly minification-compatible - _that_ name is perfectly safe to mangle / minify.

Not that it saves _much_ space, but it's something.

Reply:
> Two more references exist later on within the function, making three total. I've generally preferred to avoid constant `this.` prefixing for the same thing repeatedly when I can help it.

I understand, but it does add an extra variable, increasing the complexity of the function and potential for future error. It makes a reviewer question why there must be a local copy of the variable (i.e. is there some secret reason why we must track the value locally? does the `this` version change so we have to keep it locally? is this copy-by-value or by-reference -- do changes propagate to the instance member? what happens if one version gets renamed in the future? can't tell at a glance if two identically named variables refer to the same thing. etc.) so generally IMO it's a bit of an anti-pattern.

> Also, though but a minor point, in my experience `esbuild` doesn't minify things accessed by property - it won't minify `this.toKey`. On the other hand a local constant variable (like our copied `toKey` variable) is perfectly minification-compatible - _that_ name is perfectly safe to mangle / minify.
>
> Not that it saves _much_ space, but it's something.

I know I am all about space reduction, but I think this is probably optimizing at the wrong level. We should try to avoid writing code in order to satisfy inadequacies in tooling -- and this is especially true for optimizations.

Reply:
I'd argue the same could apply to `const root`... although there, it actually _is_ necessary in order for TS type-inference to work properly.

It _does_ apply to `const totalWeight` (a closely following line) in full, so I'll be changing it as well.

Reply:
> I'd argue the same could apply to `const root`... 
> 
> It _does_ apply to `const totalWeight` (a closely following line) in full, so I'll be changing it as well.

Oh for sure :grin: 

> although there, it actually _is_ necessary in order for TS type-inference to work properly.

Notice that means we are writing code to satisfy the tool... 😭 I know it often can't be avoided -- we just do the best we can.